src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,287 @@
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT id from drivers where id = ? and deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                carList.addAll(getCarsByIdDrivers(connection,
+                        resultSet.getObject("id", Long.class)));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+        return carList;
+    }
+
+    private void createCarsDriversDB(Car car) {
+        String queryForCarsDrivers = "INSERT INTO cars_drivers (driver_id, car_id) values (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementOfCarsDrivers = connection
+                        .prepareStatement(queryForCarsDrivers)) {
+            Iterator<Driver> iterator = car.getDrivers().iterator();
+            while (iterator.hasNext()) {
+                preparedStatementOfCarsDrivers.setLong(1, iterator.next().getId());
+                preparedStatementOfCarsDrivers.setLong(2, car.getId());
+                preparedStatementOfCarsDrivers.executeUpdate();
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't create data "
+                    + car + " into cars_drivers table", exception);
+        }
+    }
+
+    private void removeCarsDrivers(Connection connection, Car car) {
+        String queryRemove = "delete from cars_drivers where car_id = ?;";
+
+        try (PreparedStatement removeStmt = connection
+                .prepareStatement(queryRemove)) {
+            removeStmt.setLong(1, car.getId());
+            removeStmt.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't remove data of "
+                    + car + " into cars_drivers", exception);
+        }
+    }
+
+    private void insertCarsDrivers(Connection connection, Car car) {
+        String queryInsert = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?);";
+
+        try (PreparedStatement insertStmt = connection
+                     .prepareStatement(queryInsert)) {
+            Iterator<Driver> iterator = car.getDrivers().iterator();
+            while (iterator.hasNext()) {
+                insertStmt.setLong(1, iterator.next().getId());
+                insertStmt.setLong(2, car.getId());
+            }
```

Comment:
what's the point if the statement is not executed?

Reply:
I added check on 1 or 0